### ISSUE TITLE
iterate over subdirectories of extra_src or add a flag to indicate adding source archive under a directory in its own layer.

### ISSUE THREAD

**Issue by hjmodi** [containers/BuildSourceImage#79]
Currently, if we have many source archives we want to add layer by layer to the source image the only way to do that is to put source archives in their own subdirectory and pass each subdirectory's absolute path with EXTRA_SOURCE_DIR.

But as @MartinBasti pointed out in https://github.com/containerbuildsystem/atomic-reactor/pull/1606#discussion_r609879731, if we do that then there's a good chance we may reach the limit length for a command in the shell.

Adding the ability to iterate over the subdirectories automatically and add them all layer by layer or adding a flag to indicate adding each source archive in a different layer will help solve this problem.

**Comment by vrothberg** [comment]
Thanks for reaching out.

We're currently working on a [prototype in Buildah](https://github.com/containers/buildah/pull/3101).  Source images are then OCI artifacts and the tool should be easier to maintain and extend.  Your request is interesting and I will consider it.

Unless there's a very urgent need to extend BuildSourceImage, I'd prefer to add the request to Buildah.

**Comment by vrothberg** [comment]
FYI @rhatdan

**Comment by rhatdan** [comment]
I agree, lets move this package to archive and continue work in buildah.
